### PR TITLE
Jd admin2

### PR DESCRIPTION
**Pull request by josephdowns**
Add top five customers to admin dashboard

### CODE REVIEW

File: app/controllers/admin_controller.rb [hunk 1]
```diff
@@ -1,11 +1,17 @@
 class AdminController < ApplicationController
   def index
+    @customers = Customer.top_five_customers
+    binding.pry
```
Comment (jhennerich) [inline app/controllers/admin_controller.rb]: need to remove pry here

**Review by jhennerich — CHANGES_REQUESTED** [review]
need to remove the pry before merge, otherwise all tests pass on my local install.

File: app/controllers/admin_controller.rb (hunk 1 above)
Comment (josephdowns) [reply app/controllers/admin_controller.rb]: Removed!

**Comment by josephdowns** [comment]
I removed the pry

File: app/models/customer.rb [hunk 2]
```diff
@@ -3,4 +3,18 @@ class Customer < ApplicationRecord
...
   has_many :invoices
+  has_many :transactions, through: :invoices
+
+  def self.top_five_customers
+    joins(:transactions)
+    .where("result = 'success'")
+    .group(:id)
+    .select("customers.*, count('transactions.result') as transaction_count")
+    .order(transaction_count: :desc)
+    .limit(5)
```
Comment (ross-ian28) [inline app/models/customer.rb:14]: What an amazing looking method

**Review by ross-ian28 — APPROVED** [review]

**Review by Josenecal — APPROVED** [review]
No issues from my perspective!

**Pull request merged**